lib/spack/spack/cmd/install.py

```diff
@@ -246,9 +246,18 @@ def install(parser, args, **kwargs):
         # then install the packages from it.
         env = ev.get_env(args, 'install')
         if env:
+            if args.test == 'all' or args.run_tests:
+                tests = True
+            elif args.test == 'root':
+                tests = [spec.name for spec in env.user_specs]
+            else:
+                tests = False
+
+            kwargs['tests'] = tests
+
```

Comment:
Can this segment be refactored  to be earlier in the method so it can be combined with the identical segment below?